test/python/result/test_result.py

```diff
@@ -191,6 +191,21 @@ def test_marginal_counts_result(self):
         self.assertEqual(marginal_counts(result, [0, 1]).get_counts(0), expected_marginal_counts_1)
         self.assertEqual(marginal_counts(result, [0]).get_counts(1), expected_marginal_counts_2)
 
+    def test_marginal_counts_result_memory(self):
```

Comment:
Besides testing the common case we should test some additional edge cases with memory in the result. Like what if `inplace=True`, the indices list is not set or has values larger than classical register, etc?